### PR TITLE
feat: REST endpoints for metadata API

### PR DESCRIPTION
**Pull request by AdityaPimpalkar**
closes - #3648

Feedback is appreciated! :)

### CODE REVIEW

File: packages/twenty-server/src/core/api-rest/api-rest-query-builder/utils/parse-path.utils.ts [hunk 1]
```diff
@@ -19,3 +19,47 @@ export const parsePath = (
...
+): { object: string; id?: string } => {
+  const queryAction = request.path.replace('/rest/metadata/', '').split('/');
+
+  if (queryAction.length > 3) {
+    throw new BadRequestException(
+      `Query path '${request.path}' invalid. Valid examples: /rest/metadata/fields or /rest/metadata/objects/id`,
+    );
+  }
+
+  if (!['fields', 'objects', 'relations'].includes(queryAction[0])) {
```
Comment (AdityaPimpalkar) [inline packages/twenty-server/src/core/api-rest/api-rest-query-builder/utils/parse-path.utils.ts:34]: Would there be a way to somehow fetch these values directly from the db ? I don't feel comfortable hardcoding it. Or maybe I am missing something and we shouldn't do it like this

Comment (martmull) [reply packages/twenty-server/src/core/api-rest/api-rest-query-builder/utils/parse-path.utils.ts:34]: looks like hardcoding it is the best we can do for now

**Review by martmull — CHANGES_REQUESTED** [review]
Hey, thank you for your PR. Added some comments

**Comment by martmull** [comment]
Hey @AdityaPimpalkar did you take a look at requested changes? Feel free to comment if needed

**Comment by AdityaPimpalkar** [comment]
Thank you for the feedback @martmull! Appreciate it!

> Think we should have a get verb for this endpoint, like with graphql. Why ignoring it?

I was experimenting if I can dynamically add the return parameters for the GET endpoint like it's done in the graphql endpoint. But it got very complex very quickly so I thought we can hard code the return parameters for now and right the dynamic logic in the future. 

I made the changes, let me know your thoughts :)

**Comment by martmull** [comment]
Hey @AdityaPimpalkar, thank you for your response. Indeed I think we cannot do something better but hardcoding the graphql request for metadatas. I will check the PR today. Keep you informed

**Review by martmull — APPROVED** [review]

**Comment by martmull** [comment]
Really nice thank you!

**Comment by martmull** [comment]
@AdityaPimpalkar , I approved, but then i checked and there is a small issue. When requesting objects you only get 10 objects maximum in the response. I think you should add `paging: { first: 1000 }` line  like it is done in the front request (see `packages/twenty-front/src/modules/object-metadata/graphql/queries.ts`)
Also it would be great to return the fields when you return the objects (like it is done in the front request)
I will add a last commit to pass those changes

**Comment by FelixMalfait** [comment]
Quick followup issue https://github.com/twentyhq/twenty/issues/4089

**Pull request merged**